torch/utils/cpp_extension.py

```diff
@@ -978,6 +988,19 @@ def CUDAExtension(name, sources, *args, **kwargs):
     Currently open issue for nvcc bug: https://github.com/pytorch/pytorch/issues/69460
     Complete workaround code example: https://github.com/facebookresearch/pytorch3d/commit/cb170ac024a949f1f9614ffe6af1c38d972f7d48
 
+    Relocatable device code linking:
+
+    This extension supports the extra linking step needed for relocatable device code by setting ``dlink`` to ``True``.
+    Relocatable device code libraries can optionally be passed via ``dlink_libraries``.
+    Extra device linking flags (e.g. ``-dlto``) can be passed as a list via the ``nvcc_dlink`` key of ``extra_compile_args``.
```

Comment:
It's not clear to me from the docs here why I'd want to enable relocatable device code.